src/main/java/org/bahmni/reports/template/TSIntegrationDiagnosisReportTemplate.java

```diff
@@ -0,0 +1,145 @@
+package org.bahmni.reports.template;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import net.sf.dynamicreports.jasper.builder.JasperReportBuilder;
+import net.sf.dynamicreports.report.constant.HorizontalAlignment;
+import net.sf.dynamicreports.report.constant.PageType;
+import net.sf.dynamicreports.report.constant.WhenNoDataType;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.bahmni.reports.model.Report;
+import org.bahmni.reports.model.TSIntegrationDiagnosisReportConfig;
+import org.bahmni.reports.model.TSPageObject;
+import org.bahmni.reports.model.UsingDatasource;
+import org.bahmni.reports.report.BahmniReportBuilder;
+import org.bahmni.reports.util.CommonComponents;
+import org.bahmni.webclients.HttpClient;
+import org.stringtemplate.v4.ST;
+
+import java.io.IOException;
+import java.net.URI;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.text.MessageFormat;
+import java.util.List;
+import java.util.Properties;
+
+import static net.sf.dynamicreports.report.builder.DynamicReports.col;
+import static net.sf.dynamicreports.report.builder.DynamicReports.type;
+import static org.bahmni.reports.template.Templates.minimalColumnStyle;
+import static org.bahmni.reports.util.FileReaderUtil.getFileContent;
+
+
+@UsingDatasource("openmrs")
+public class TSIntegrationDiagnosisReportTemplate extends BaseReportTemplate<TSIntegrationDiagnosisReportConfig> {
+    public static final String CREATE_SQL_TEMPLATE = "create temporary table {0}(code varchar(100) not null)";
```

Comment:
What does this code do? Create a temporary table in DB, and store some rows, and later delete it when report is generated? The code could not store this data in-memory?

Reply:
For every report request being made, it creates one 'unique' temp table which will be active for the duration of the session. Based on the parent SNOMED code, it makes paginated calls to Snowstorm server and collects all the descendant codes till it exhausts the list. While collecting SNOMED codes each time, it pushes them into the temp table, so that in-memory is not overloaded. Also the inner join with the temporary table offers better performance versus passing from in-memory. The temporary table gets automatically purged once the session is over.